include/slb/detail/config.hpp

```diff
@@ -27,4 +27,24 @@
 #define __has_feature(x) 0
 #endif
 
+// libstdc++ did not mark `operator()` as `noexcept` until version 8.
+#if __cpp_lib_integral_constant_callable &&                                    \
+    (!defined(__GLIBCXX__) || __has_include(<filesystem>))
+#define SLB_CPP_LIB_INTEGRAL_CONSTANT_CALLABLE 2 // available / conforming
+#else
+#define SLB_CPP_LIB_INTEGRAL_CONSTANT_CALLABLE 0 // not available
+#endif
+
+// libstdc++ did not mark `size()` as `noexcept` until version 8.
+#if __cpp_lib_integer_sequence ||                                              \
+    (_LIBCPP_VERSION >= 1101 && _LIBCPP_STD_VER > 11) || defined(_MSC_VER)
+#if defined(__GLIBCXX__) && __has_include(<filesystem>)
+#define SLB_CPP_LIB_INTEGER_SEQUENCE 2 // available / conforming
+#else
+#define SLB_CPP_LIB_INTEGER_SEQUENCE 1 // available / non-conforming
+#endif
+#else
+#define SLB_CPP_LIB_INTEGER_SEQUENCE 0 // not available
+#endif
+
 #endif // SLB_DETAIL_CONFIG_HPP
```

Comment:
SD-6 library macros are only defined when their corresponding headers have been included. The values of the `__cpp_lib_integral_constant_callable` and `__cpp_lib_integer_sequence` checks here will depend too on whether the corresponding headers have been included previously to including `config.hpp`, which could further be different across TUs.

Placing these definitions in `type_traits.hpp` and `utility.hpp` correspondingly instead would guarantee that the necessary includes are in place, and would make them available to anyone who could need them.

Reply:
Oh wow. You're right... :(